src/controllers/insertIssueTable.gs

```diff
@@ -0,0 +1,268 @@
+
+/**
+ * @file Contains controller class and dialog/callback method for inserting issue tables from jira filter.
+ */
+
+/**
+ * @desc Wrapper: Dialog to choose issues filter
+ */
+function menuInsertIssueFromFilter() {
+  InsertIssueTable_Controller_.dialogOpen();
+}
+
+/**
+ * @desc Wrapper: Dialog callback handler
+ * @return {object} Object({status: [boolean], response: [string]})
+ */
+function callbackInsertIssueFromFilter(jsonFormData) {
+  return InsertIssueTable_Controller_.callbackInsertFromFilter(jsonFormData);
+}
+
+/**
+ * Creates a new IssueTableIndex_ object, which is used to persist IssueTables and related information.
+ */
+InsertIssueTable_Controller_ = {
+  name : 'InsertIssueTable_Controller_',
+
+  /**
+   * @desc Dialog to configure Jira custom fields
+   */
+  dialogOpen : function () {
+    debug.log(this.name + '.dialog()');
+
+    if (!hasSettings(true))
+      return;
+
+    // prune table and check if user tries to overwrite/overlap with an existing IssueTable
+    IssueTableIndex_.prune();
+    if (true === this.hasOverlappingTable(true) ) {
+      // will show warning/alter and skip opening the dialog
+      return;
+    }
+
+    var customFields = IssueFields.getAvailableCustomFields(IssueFields.CUSTOMFIELD_FORMAT_SEARCH);
+    var userColumns = UserStorage.getValue('userColumns') || [];
+    var dialog = getDialog('views/dialogs/insertIssueFromFilter', {
+      columns : IssueFields.getBuiltInJiraFields(),
+      customFields : customFields,
+      userColumns : userColumns.length > 0 ? userColumns : jiraColumnDefault
+    });
+
+    // try to adjust height depending on amount of jira fields to show
+    var rowH = 32;
+    var height = 424;
+    height += (Math.ceil(Object.keys(IssueFields.getBuiltInJiraFields()).length % 4) * rowH);
+    height += (Math.ceil(Object.keys(customFields).length % 4) * rowH);
+
+    dialog
+      .setWidth(600)
+      .setHeight(height)
+      .setSandboxMode(HtmlService.SandboxMode.IFRAME);
+
+    debug.log('Processed: %s', dialog);
+
+    SpreadsheetApp.getUi().showModalDialog(dialog, 'List Jira issues from filter');
+  },
+
+  /**
+   * @desc Form handler for dialogIssuesFromFilter. Retrieve issues for given filter with specified columns from Jira and insert into
+   *       current active sheet.
+   * @param jsonFormData {object} JSON Form object of all form values
+   * @return {object} Object({status: [boolean], response: [string]})
+   */
+  callbackInsertFromFilter : function (jsonFormData) {
+    debug.log(this.name + '.callbackInsertFromFilter() <= %s', JSON.stringify(jsonFormData));
+
+    jsonFormData = jsonFormData || {filter_id: 0};
+    var that = this,
+        startAt = parseInt(jsonFormData['startAt']) || 0,
+        columns = jsonFormData['columns'] || jiraColumnDefault,
+        response = {status: false, message: ''};
+
+    var Renderer, attributes = {
```

Comment:
see comment below about Renderer in the if statement.
This shouldnt be initialised this early - should be initialised closer to the scope its used in 
Setting it here implies its used throughout the scope of the function

Reply:
Yeah you're right. I always struggle between the style guides telling to initialize/define all variables in a block at beginning instead of in between of all the code logic...